fstar-helpers/core-models/src/core_arch/x86/interpretations.rs

```diff
@@ -28,18 +36,509 @@ pub mod int_vec {
+
+    pub fn _mm_sllv_epi32(a: i32x4, b: i32x4) -> i32x4 {
+        i32x4::from_fn(|i| {
+            if b[i] > 31 || b[i] < 0 {
+                0
+            } else {
+                ((a[i] as u32) << b[i]) as i32
+            }
+        })
+    }
+
+    pub fn _mm256_slli_epi64<const IMM8: i32>(a: i64x4) -> i64x4 {
+        i64x4::from_fn(|i| {
+            let imm8 = IMM8 % 256;
+            if imm8 > 63 {
+                0
+            } else {
+                ((a[i] as u64) << imm8) as i64
+            }
+        })
+    }
+
+    pub fn _mm256_bsrli_epi128<const IMM8: i32>(a: i128x2) -> i128x2 {
+        i128x2::from_fn(|i| {
+            let tmp = IMM8 % 256;
+            let tmp = tmp % 16;
+            ((a[i] as u128) >> (tmp * 8)) as i128
+        })
+    }
+
+    pub fn _mm256_andnot_si256(a: BitVec<256>, b: BitVec<256>) -> BitVec<256> {
+        BitVec::from_fn(|i| match (a[i], b[i]) {
+            (Bit::Zero, Bit::One) => Bit::One,
+            _ => Bit::Zero,
+        })
+    }
+
+    pub fn _mm256_set1_epi64x(a: i64) -> i64x4 {
+        i64x4::from_fn(|_| a)
+    }
+
+    pub fn _mm256_set_epi64x(e3: i64, e2: i64, e1: i64, e0: i64) -> i64x4 {
+        i64x4::from_fn(|i| match i {
+            0 => e0,
+            1 => e1,
+            2 => e2,
+            3 => e3,
+            _ => unreachable!(),
+        })
+    }
+
+    pub fn _mm256_unpacklo_epi64(a: i64x4, b: i64x4) -> i64x4 {
+        i64x4::from_fn(|i| match i {
+            0 => a[0],
+            1 => b[0],
+            2 => a[2],
+            3 => b[2],
+            _ => unreachable!(),
+        })
+    }
+
+    pub fn _mm256_permute2x128_si256<const IMM8: i32>(a: i128x2, b: i128x2) -> i128x2 {
+        i128x2::from_fn(|i| {
+            let control = IMM8 >> (i * 4);
+            if (control >> 3) % 2 == 1 {
+                0
+            } else {
+                match control % 4 {
+                    0 => a[0],
+                    1 => a[1],
+                    2 => b[0],
+                    3 => b[1],
+                    _ => unreachable!(),
+                }
+            }
+        })
     }
 
     pub use lemmas::flatten_circuit;
-    // ! This module provides lemmas allowing to lift the intrinsics modeled in `super` from their version operating on AVX2 vectors to functions operating on machine integer vectors (e.g. on `i32x8`).
     pub mod lemmas {
+        //! This module provides lemmas allowing to lift the intrinsics modeled in `super` from their version operating on AVX2 vectors to functions operating on machine integer vectors (e.g. on `i32x8`).
+        #[cfg(hax)]
         use super::*;
 
-        #[allow(unused)]
+        #[cfg(hax)]
         use crate::core_arch::x86 as upstream;
-        #[allow(unused)]
-        use crate::core_arch::x86::__m256i;
+        #[cfg(hax)]
+        use crate::core_arch::x86::{__m128i, __m256, __m256i};
 
         /// An F* attribute that marks an item as being an lifting lemma.
         #[allow(dead_code)]
```

Comment:
Why allow dead code? That should never be necessary. Either it's public and could be used by a consumer, or it's not used, that it should go away.